src/metadata.rs

```diff
@@ -27,6 +27,7 @@ pub trait Metadata: MajorObject {
                 domains.push(_string(p));
             }
         }
+        unsafe { gdal_sys::CSLDestroy(c_res) };
```

Comment:
Shouldn't this be inside the not-null block or is it okay to call this function if `c_res` is null?

Reply:
Frees the passed string list (null terminated array of strings). It is safe to pass NULL.
https://gdal.org/doxygen/ogr__api_8h.html#a4000d426bf26ad7cc7d4012634c93f09

Reply:
The correct link is https://gdal.org/doxygen/cpl__string_8h.html#a5a39b9c5896a273cc6f06c1d5be93238.
Okay, seems fine with me. Again, I cannot resolve this comment.